telemetry/src/metrics.rs

```diff
@@ -0,0 +1,174 @@
+            Opts::new("isi", "Iroha special instructions handled by this peer"),
+            &["type", "success_status"],
+        )
+        .expect("Infallible");
+        let isi_times = HistogramVec::new(
+            HistogramOpts::new("isi_times", "Time to handle isi in this peer"),
+            &["type"],
+        )
+        .expect("Infallible");
+        let tx_amounts = Histogram::with_opts(HistogramOpts::new(
+            "tx_amount",
+            "average amount involved in a transaction on this peer",
+        ))
+        .expect("Infallible");
+        let block_height =
+            IntCounter::new("block_height", "Current block height").expect("Infallible");
+        let connected_peers = GenericGauge::new(
+            "connected_peers",
+            "Total number of currently connected peers",
+        )
+        .expect("Infallible");
+        let uptime_since_genesis_ms = GenericGauge::new(
+            "uptime_since_genesis_ms",
+            "Network up-time, from creation of the genesis block",
+        )
+        .expect("Infallible");
+        let domains = GenericGauge::new("domains", "Total number of domains").expect("Infallible");
+        let accounts = GenericGaugeVec::new(
+            Opts::new("accounts", "User accounts registered at this time"),
+            &["domain"],
+        )
+        .expect("Infallible");
+        let registry = Registry::new();
+
+        macro_rules! register {
+			($metric:expr)=> {
+				registry.register(Box::new($metric.clone())).expect("Infallible");
+			};
+			($metric:expr,$($metrics:expr),+)=>{
+				register!($metric);
+				register!($($metrics),+);
+			}
+		}
+
+        register!(
+            txs,
+            tx_amounts,
+            block_height,
+            connected_peers,
+            uptime_since_genesis_ms,
+            domains,
+            accounts,
+            isi,
+            isi_times
+        );
+
+        Self {
+            txs,
+            block_height,
+            connected_peers,
+            uptime_since_genesis_ms,
+            registry,
+            domains,
+            accounts,
+            tx_amounts,
+            isi,
+            isi_times,
+        }
+    }
+}
+
+impl Metrics {
+    /// Convert the current [`Metrics`] into a Prometheus-readable format.
+    ///
+    /// # Errors
+    /// - If [`Encoder`] fails to encode the data
+    /// - If the buffer produced by [`Encoder`] causes [`String::from_utf8`] to fail.
+    pub fn try_to_string(&self) -> eyre::Result<String> {
+        let mut buffer = vec![];
+        let encoder = prometheus::TextEncoder::new();
+        let metric_families = self.registry.gather();
+        Encoder::encode(&encoder, &metric_families, &mut buffer)?;
+        Ok(String::from_utf8(buffer)?)
+    }
+
+    /// Get time elapsed since Unix epoch.
+    ///
+    /// # Panics
+    /// Never
+    #[allow(clippy::unused_self, clippy::expect_used)]
+    pub fn current_time(&self) -> Duration {
```

Comment:
make it associated function? also, maybe would make more sense to make this a freestanding method of `telemetry`?

Reply:
This is a dirty hack that means that a user of `iroha_telemetry_derive` doesn't also need to depend on `iroha_telemetry`. This also means that instead of inlining this function into the macro, I get to call a neat `metrics.current_time()`.